HIP/hip-1056.md

```diff
@@ -182,34 +185,48 @@ over the stream.
-over the entire history of the stream. Of course, older blocks can be represented by their hash rather than the actual
-full contents, so it is possible to have a very large block stream merkle tree *in concept* that is still very efficient
-to use.
+At each block boundary, the state merkle tree is a depth two *subtree* of a `block merkle tree`. 
+The block merke tree is a binary merkle tree whose four subtrees at depth two also include the block's input items,
+the block's output items, and the merkle tree of the previous block. 
+Of course Hash computation for the block `N` merkle tree reuses the hash computed for the block `N-1` merkle tree.
+
+> Merkle Item Order vs Block Item Order: The order in which items appear in the block merkle tree are not the same 
+> order in which are streamed. The left-to-right order of merkle leaves must be the order in which those items are
+> encountered in the stream. Some items are in the "input" subtree and some are in the "output" subtree, within those
+> trees the order will be the same as stream-order.
+
+
+> State Merkle Hash: A block contains the state hash that represents the state at the beginning of the block. 
+> The hash of the state at the end of the block is found in the next block. This is because it takes time to gather state
+> changes of the block and hash them, waiting for this process would result in the block delivery time being held up.
+> Rapid block times is a goal of the block stream and thus a tradeoff is made to ensure minimum latency. 
+> The impact of this choice is that a state proof for verification of block `N` requires block `N+1`.
+> However, the state proof can still be calculated by all the data contained in block `N`. A block is self contained!
 
 #### Design for Verifiability
 
 Existing record files contain the following cryptographic properties.
 
-Record files are streams in a blockchain, with the hash of previous file included in the next record file, ensuring
-immutability. Additionally, each consensus node on the Hedera mainnet signs a hash of the record files as they are
+Record file contents are summarized in a chain of hashes, with each file's contents containing the hash of the previous
+file so that the hash of the current file depends on the contents of the entire record stream.
+Additionally, each consensus node on the Hedera mainnet signs a hash of the record files as it is
 produced, enabling mirror nodes and other downstream services to verify that the record file was produced and attested
-by a majority weight of the network nodes. A major drawback of the current design is that multiple signature files and
-the address book with network weight must be collected and verified before a record file can be verified. This design
+by a majority stake weight of the network nodes. A major drawback of the current design is that multiple signature files and
```

Comment:
I know I've said this before, but using the term "stake weight" is very strange. There stake, and there is weight. Stake is measured in hbars. Weight is unit-less and determined from total network stake. If we mean weight, just say "weight". But maybe there has been an agreement to use "stake weight" that I'm not aware of?

Reply:
Hmm, there was a correction to utilize stake weight.
Let me confirm, @jsync-swirlds thoughts?

Reply:
I think using stake weight everywhere is fine; we should make sure it's defined however.

Reply:
Does that mean we will have 3 terms, `stake`, `weight`, and `stake weight` where the last two are equivalent? These seems exceedingly confusing.